nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -591,6 +647,22 @@ def setup(self, stage=None):
         Args:
             stage (str, optional): Can be 'fit', 'validate', 'test' or 'predict'. Defaults to None.
         """
+
+        # log number of parameters
+        if self.is_data_parallel_rank_zero():
+            if isinstance(self.model, list):
+                num_parameters = sum(
+                    [sum([p.nelement() for p in model_module.parameters()]) for model_module in self.model]
+                )
+            else:
+                num_parameters = sum([p.nelement() for p in self.model.parameters()])
+
+            logging.info(
+                f'Pipeline model parallel rank: {parallel_state.get_pipeline_model_parallel_rank()}, '
+                f'Tensor model parallel rank: {parallel_state.get_tensor_model_parallel_rank()}, '
+                f'Number of parameters: {num_parameters:.2e}.'
```

Comment:
do we need to do all reduce on this num_parameters?

Reply:
Good point. I think we can log both num_parameters per model parallel rank and then log the total number per model

Reply:
done